packages/cli/commands/sandbox/delete.js

```diff
@@ -2,22 +2,122 @@ const {
 const { loadAndValidateOptions } = require('../../lib/validation');
+
+const { deleteSandbox } = require('@hubspot/cli-lib/sandboxes');
 const { i18n } = require('@hubspot/cli-lib/lib/lang');
+const { getConfig } = require('@hubspot/cli-lib');
+const { deleteSandboxPrompt } = require('../../lib/prompts/sandboxesPrompt');
+const { removeAccountFromConfig } = require('@hubspot/cli-lib/lib/config');
+const {
+  selectAndSetAsDefaultAccountPrompt,
+} = require('../../lib/prompts/accountsPrompt');
+const { EXIT_CODES } = require('../../lib/enums/exitCodes');
+const { promptUser } = require('../../lib/prompts/promptUtils');
 
 const i18nKey = 'cli.commands.sandbox.subcommands.delete';
 
-exports.command = 'delete';
+const SANDBOX_NOT_FOUND = 'SandboxErrors.SANDBOX_NOT_FOUND';
+const OBJECT_NOT_FOUND = 'OBJECT_NOT_FOUND';
+
+exports.command = 'delete [--account]';
 exports.describe = i18n(`${i18nKey}.describe`);
 
 exports.handler = async options => {
-  await loadAndValidateOptions(options);
+  await loadAndValidateOptions(options, false);
+
+  const { account } = options;
+  const config = getConfig();
+
+  let accountPrompt;
+  if (!account) {
+    accountPrompt = await deleteSandboxPrompt(config);
+  }
+  const sandboxAccountId = getAccountId({
+    account: account || accountPrompt.account,
+  });
+
+  trackCommandUsage('sandbox-delete', {}, sandboxAccountId);
+
+  let parentAccountId;
+  for (const portal of config.portals) {
+    if (portal.portalId === sandboxAccountId) {
+      if (portal.parentAccountId) {
+        parentAccountId = portal.parentAccountId;
+      } else {
+        const parentAccountPrompt = await deleteSandboxPrompt(config, true);
+        parentAccountId = getAccountId({
+          account: parentAccountPrompt.account,
+        });
+      }
+    }
+  }
+
+  logger.debug(
+    i18n(`${i18nKey}.debug.deleting`, {
+      account: account || accountPrompt.account,
+    })
+  );
+
+  try {
+    const { confirmSandboxDeletePrompt: confirmed } = await promptUser([
+      {
+        name: 'confirmSandboxDeletePrompt',
+        type: 'confirm',
+        message: i18n(`${i18nKey}.confirm`, {
+          account: account || accountPrompt.account,
+        }),
+      },
+    ]);
+    if (!confirmed) {
+      process.exit(EXIT_CODES.SUCCESS);
+    }
+
+    await deleteSandbox(parentAccountId, sandboxAccountId);
+
+    logger.log('');
+    logger.success(
+      i18n(`${i18nKey}.success.delete`, {
+        account: account || accountPrompt.account,
+        sandboxHubId: sandboxAccountId,
+      })
+    );
+    logger.log('');
+
+    const promptDefaultAccount = removeAccountFromConfig(sandboxAccountId);
+    if (promptDefaultAccount) {
+      await selectAndSetAsDefaultAccountPrompt(config);
+    }
+    process.exit(EXIT_CODES.SUCCESS);
+  } catch (err) {
+    if (
+      err.error &&
+      err.error.category === OBJECT_NOT_FOUND &&
+      err.error.subCategory === SANDBOX_NOT_FOUND
```

Comment:
This is if a user deletes a sandbox from the UI and it still exists in the CLI config. 

I also bypassed the account validation on delete because I was getting errors that the PAK no longer was valid (account was already deleted), which was correct but that was preventing users from removing accounts.